### PR TITLE
Move onboot startup script to runc package

### PR DESCRIPTION
**Pull request by justincormack**
As onboot does not use containerd at all, this means you can run very
    minimal setups with just `runc` if you use no services, for example
    most of our tests do not actually use services, or if you have other
    similar very minimal use cases.
    
    Move ulimit setup to `init` which makes more sense.

frog

### CODE REVIEW

File: projects/compose/compose-dynamic.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ kernel:
   cmdline: "console=ttyS0 page_poison=1"
 init:
   - linuxkit/init:12348442d56c2ee9abf13ff38dff2e36b515bd1e
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:f73ab00a4d18edf6beb056685426e9d6b5e3d254
   - linuxkit/containerd:b1766e4c4c09f63ac4925a6e4612852a93f7e73b
```
Comment (rn) [inline projects/compose/compose-dynamic.yml]: missed this update

File: projects/compose/compose-static.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ kernel:
   cmdline: "console=ttyS0 page_poison=1"
 init:
   - linuxkit/init:12348442d56c2ee9abf13ff38dff2e36b515bd1e
-  - linuxkit/runc:2310ad9d266cf5d9c4d07613bd2135ed7eb8a21f
+  - linuxkit/runc:f73ab00a4d18edf6beb056685426e9d6b5e3d254
   - linuxkit/containerd:b1766e4c4c09f63ac4925a6e4612852a93f7e73b
```
Comment (rn) [inline projects/compose/compose-static.yml]: and here

File: projects/etcd/etcd.yml [hunk 3]
```diff
@@ -3,8 +3,8 @@ kernel:
   cmdline: "console=ttyS0 console=tty0 page_poison=1"
 init:
   - linuxkit/init:12348442d56c2ee9abf13ff38dff2e36b515bd1e
```
Comment (rn) [inline projects/etcd/etcd.yml]: init needs an update

File: projects/ima-namespace/ima-namespace.yml [hunk 4]
```diff
@@ -3,8 +3,8 @@ kernel:
   cmdline: "console=ttyS0 console=tty0 page_poison=1 ima_appraise=enforce_ns"
 init:
   - linuxkit/init:b3740303f3d1e5689a84c87b7dfb48fd2a40a192
```
Comment (rn) [inline projects/ima-namespace/ima-namespace.yml]: and here as well

File: projects/okernel/examples/okernel_simple.yaml [hunk 5]
```diff
@@ -3,7 +3,7 @@ kernel:
   cmdline: "console=ttyS0 console=tty0 page_poison=1"
 init:
   - linuxkit/init:12348442d56c2ee9abf13ff38dff2e36b515bd1e
```
Comment (rn) [inline projects/okernel/examples/okernel_simple.yaml]: init and containerd updates missed here

File: projects/wireguard/wireguard.yml [hunk 6]
```diff
@@ -3,7 +3,7 @@ kernel:
   cmdline: "console=ttyS0 console=tty0 page_poison=1"
 init:
   - linuxkit/init:cbd7ae748f0a082516501a3e914fa0c924ee941e
```
Comment (rn) [inline projects/wireguard/wireguard.yml]: init and containerd update missed here

**Comment by justincormack** [comment]
At some times at least I have not been updating hashes in projects as they may have reasons to use things, not sure what to do there. I was about to write a script to check hashes are correct, but was going to omit projects...

**Comment by rn** [comment]
then surely we shouldn't update any hashes which some of the changes do

**Comment by justincormack** [comment]
Yes, projects are currently a mess, as some of them do have special requirements and some do not. The change above was just changing all the ones that were currently the same as the ones in the main example, rather than updating everything regardless of how it was set before, which is what I usually do on the grounds if it matches other things it cant be special. I dont think that works in all cases (eg logging probably broken) though.

**Comment by ijc** [comment]
Yeah, my MO has been to update anything which matches the toplevel `linuxkit.yml`.

**Comment by justincormack** [comment]
Added the others

**Comment by rn** [comment]
not all of them, but LGTM

**Comment by riyazdf** [comment]
re: projects updating, agree with @justincormack that it's been fairly adhoc and messy.  I'm fine with leaving them as-is for most updates.

**Comment by justincormack** [comment]
This does mean that if you mount a persistent disk under the `containerd` directory (and don't eg mount a `tmpfs` there too) then the `containerd` image store etc will be persistent.

**Comment by justincormack** [comment]
(I think this is better behaviour, previously it would probably leave a mess).

**Comment by justincormack** [comment]
(Also needed if you want to eg test `containerd` on btrfs root.)

Just removing `contaienrd` from the tests where not required.

**Pull request merged**